### PR TITLE
feat: add chromedriverExecutableDir and autodownload support

### PR DESCRIPTION
**Pull request by KazuCocoa**
Closes https://github.com/headspinio/appium-lg-webos-driver/issues/43
cc @jlipps

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +50,9 @@ package in your `package.json`)
 |`appium:rcMode`|[Optional; default `js`; must be `rc` or `js`] When the value is `js`, the `webos: pressKey` command will operate with JS executed via Chromedriver. Otherwise, keys will be sent using the websocket remote control API. Note that when `appium:remoteOnly` is set to true, the value of `appium:rcMode` will always behave as if set to `rc`.|
 |`appium:keyCooldown`|[Optional; default `750`] How long to wait in between remote key presses|
 
+(*) `appium:chromedriverExecutable` or `appium:chromedriverExecutableDir` are required. The chromedriver autodwonload works only when `appium:chromedriverExecutableDir` is provided.
+If both capabilities are given, `appium:chromedriverExecutableDir` will be prior.
```
Comment (jlipps) [inline README.md:54]:
```suggestion
If both capabilities are given, `appium:chromedriverExecutableDir` will take priority.
```

**Review by jlipps — APPROVED** [review]

**Comment by KazuCocoa** [comment]
@jlipps maybe we could create 0.3.0 with this.

**Comment by jlipps** [comment]
it's now published.

**Comment by Jiyvn** [comment]
@KazuCocoa @jlipps autodownload not work on my side
image
image

appium server --allow-insecure chromedriver_autodownload
[lgtv2019.log](https://github.com/headspinio/appium-lg-webos-driver/files/14693252/lgtv2019.log)

**Comment by KazuCocoa** [comment]
Because the embedded driver did not respond what chrome version the device needed. This implementation cares https://github.com/headspinio/appium-lg-webos-driver/pull/48/files#diff-9dd24cf3101d02ce31204899de7beaa39759860d491e54541dcc55c10ec2ae32R37-R117 cases, but not sure what chromedriver should be used for your device case.

`WebKit-Version":"537.36` is very common value as the device so it does not help to guess necessary chromedriver. What we could do here is return better error message for this `"User-Agent":""` case.



```
[38;5;48m[WebOSDriver][0m Waiting for app launch to take effect
[38;5;48m[WebOSDriver][0m Waiting for app launch to take effect
[38;5;48m[WebOSDriver][0m The response of http://192.168.10.187:9998/json/version was {"Browser":"","Protocol-Version":"1.1","User-Agent":"","WebKit-Version":"537.36 (@fa89da905405aab455e0f0d4ec7f49631c7ca70b)"}
[38;5;80m[Chromedriver@b2fd][0m Changed state to 'starting'
[38;5;80m[Chromedriver@b2fd][0m Using the static Chromedriver->Chrome mapping
[38;5;80m[Chromedriver@b2fd][0m The most recent known Chrome version: 123.0.6312
[38;5;80m[Chromedriver@b2fd][0m Found 0 executables in '/Users/jiyvn/aviu/apps/lgtv/chromedriver/'
```

**Comment by KazuCocoa** [comment]
https://github.com/headspinio/appium-lg-webos-driver/issues/50 to return better error message as no info to guess necessary chromedriver version

**Pull request merged**